Actions/Github-Helper.psm1

```diff
@@ -708,91 +708,240 @@ function Set-JsonContentLF {
+            Write-Host "- $runsURI"
+            $workflowRun = InvokeWebRequest -Headers $headers -Uri $runsURI | ConvertFrom-Json
+
+            if($workflowRun.name -ne ' CI/CD') {
+                Write-Verbose "Workflow run $($workflowRun.id) (name: $($workflowRun.name)) is not a CI/CD run"
+                continue
+            }
+
+            if($workflowRun.status -ne "completed") {
+                Write-Verbose "Workflow run $($workflowRun.id) (name: $($workflowRun.name)) is not completed"
+                continue
+            }
+
+            if($workflowRun.conclusion -ne "success") {
+                Write-Verbose "Workflow run $($workflowRun.id) (name: $($workflowRun.name)) did not succeed"
+                continue
             }
+
+            $CICDRun = $workflowRun.id
+            break
         }
-        if ($total_count -le $page_no*$per_page) {
-            # no more pages
+
+        if ($CICDRun -ne 0) {
+            # CICD run was found, breaking out of the loop
+            Write-Host "Found CICD run $CICDRun that built version $version"
             break
         }
-        $page_no += 1
+
+        $page += 1
     }
-    if ($buildOutputArtifacts.Count -eq 0) {
-        Write-Host "No matching buildOutput artifacts found"
+
+    return $CICDRun
+}
+
+<#
+    Gets the project artifacts for the specified repository, branch, mask and version.
+
+    The project artifacts are returned as an array of artifact objects.
+
+    If the version is 'latest', the artifacts from the last successful CICD run are returned.
+    Otherwise, the artifacts from the CICD run that built the specified project, mask and version are returned.
+#>
+function GetArtifacts {
+    Param(
+        [Parameter(Mandatory = $true)]
+        [string] $token,
+        [Parameter(Mandatory = $true)]
+        [string] $api_url,
+        [Parameter(Mandatory = $true)]
+        [string] $repository,
+        [Parameter(Mandatory = $true)]
+        [string] $mask,
+        [Parameter(Mandatory = $true)]
+        [string] $branch,
+        [Parameter(Mandatory = $true)]
+        [string] $projects,
+        [Parameter(Mandatory = $true)]
+        [string] $version
+    )
+
+    $headers = GetHeader -token $token
+    if ($version -eq 'latest') { $version = '*' }
+
+    # Get sanitized project names (the way they appear in the artifact names)
+    $projects = @(@($projects.Split(',')) | ForEach-Object { $_.Replace('\','_').Replace('/','_') })
+
+    $CICDrun = 0
+
+    if($version -eq '*') {
+        # For latest version, use the artifacts from the last successful CICD run
+        $CICDrun = FindLatestSuccessfulCICDRun -token $token -api_url $api_url -repository $repository -branch $branch
+    } else {
+        # For specific version, find the corresponding CICD run based on the first project
+        $CICDrun = FindCICDRunForVersion -token $token -api_url $api_url -repository $repository -branch $branch -project $project[0] -mask $mask -version $version
```

Comment:
Using _BuildOutput_ would also work but build output artifacts are produced even on failed CICDs, so that's why I went with the first project in the list.